netlogo-gui/src/main/app/Tabs.scala

```diff
@@ -189,73 +266,89 @@ class Tabs(val workspace:       GUIWorkspace,
     }
   }
 
-  def handle(e: ExternalFileSavedEvent) =
+  def handle(e: ExternalFileSavedEvent) = {
     getTabWithFilename(Right(e.path)) foreach { tab =>
-      val index = indexOfComponent(tab)
+      val index = tabManager.getTabOwner(tab).indexOfComponent(tab)
       setTitleAt(index, tab.filenameForDisplay)
       tabActions(index).putValue(Action.NAME, e.path)
     }
+  }
 
-  def getSource(filename: String): String = getTabWithFilename(Right(filename)).map(_.innerSource).orNull
+  def getSource(filename: String): String = { getTabWithFilename(Right(filename)).map(_.innerSource).orNull }
 
-  def getTabWithFilename(filename: Filename): Option[TemporaryCodeTab] =
+  def getTabWithFilename(filename: Filename): Option[TemporaryCodeTab] = {
     externalFileTabs find (_.filename == filename)
+  }
 
   private var _externalFileNum = 1
   private def externalFileNum() = {
     _externalFileNum += 1
     _externalFileNum - 1
   }
-  def newExternalFile() = addNewTab(Left(I18N.gui.getN("tabs.external.new", externalFileNum(): Integer)))
 
-  def openExternalFile(filename: String) =
+  def newExternalFile() = {
+    addNewNLSTab(Left(I18N.gui.getN("tabs.external.new", externalFileNum(): Integer)))
+  }
+
+  def openExternalFile(filename: String) = {
     getTabWithFilename(Right(filename)) match {
-      case Some(tab) => setSelectedComponent(tab)
-      case _ => addNewTab(Right(filename))
+      case Some(tab) => tabManager.setPanelsSelectedComponent(tab)
+      case _ => addNewNLSTab(Right(filename))
     }
+  }
 
-  def addNewTab(name: Filename) = {
-    val tab = new TemporaryCodeTab(workspace, this, name, externalFileManager, fileManager.convertTabAction _, codeTab.smartTabbingEnabled)
+  def addNewNLSTab(name: Filename) = {
```

Comment:
Don't we call them "External Files" in the rest of the code?  Would it be `addNewExternalFileTab`?